test/integration/userMigrations.test.js

```diff
@@ -0,0 +1,71 @@
+const chai = require("chai");
+const { expect } = chai;
+const chaiHttp = require("chai-http");
+
+const app = require("../../server");
+const authService = require("../../services/authService");
+const addUser = require("../utils/addUser");
+const cleanDb = require("../utils/cleanDb");
+// Import fixtures
+const userData = require("../fixtures/user/user")();
+const superUser = userData[4];
+const nonSuperUser = userData[0];
+const colorBearingUsernames = [superUser.username, nonSuperUser.username];
+
+const config = require("config");
+const cookieName = config.get("userToken.cookieName");
+
+chai.use(chaiHttp);
+
+describe("userColorMigrations", function () {
+  let superUserId;
+  let superUserAuthToken;
+  let userId = "";
+  let nonSuperUserId = "";
+  beforeEach(async function () {
+    userId = await addUser(nonSuperUser);
+    superUserId = await addUser(superUser);
+    nonSuperUserId = userId;
+    superUserAuthToken = authService.generateAuthToken({ userId: superUserId });
+  });
+
+  afterEach(async function () {
+    await cleanDb();
+  });
+
+  describe("PATCH /migrations/user-default-color", function () {
+    it("Should return 401 if user is not a super user", function (done) {
+      const nonSuperUserJwt = authService.generateAuthToken({ userId: nonSuperUserId });
+      chai
+        .request(app)
+        .patch(`/migrations/user-default-color`)
+        .set("cookie", `${cookieName}=${nonSuperUserJwt}`)
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
+          expect(res).to.have.status(401);
+          expect(res.body).to.be.a("object");
+          expect(res.body.message).to.equal("You are not authorized for this action.");
+          return done();
+        });
+    });
+    it("Should add default color property to all users,using authorized user (super_user)", function (done) {
+      chai
+        .request(app)
+        .patch(`/migrations/user-default-color`)
+        .set("cookie", `${cookieName}=${superUserAuthToken}`)
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
+
+          expect(res).to.have.status(200);
+          expect(res.body.usersDetails.totalUsersFetched).to.be.equal(colorBearingUsernames.length);
+          expect(res.body.usersDetails.totalUsersUpdated).to.be.equal(colorBearingUsernames.length);
+          expect(res.body.usersDetails.totalUsersUnaffected).to.be.equal(0);
```

Comment:
can we also write a test in which we test users who don't have the color property pre-existing are not affected.

Reply:
sure , i had that check, but after removing usernames i removed it, ill add it back in some other. way

Reply:
@isVivek99 still can't see the assertion for users who have the color property pre-existing and are not affected.

Reply:
i have added a similar test in the model test.
[test/unit/models/userMigrations.test.js](https://github.com/Real-Dev-Squad/website-backend/pull/735/files#diff-706d009a498138cfe882b0ee3428b8357316e116c8f4866382eb4ea1f328d6da)
does this suffice?

Reply:
> i have added a similar test in the model test. [test/unit/models/userMigrations.test.js](https://github.com/Real-Dev-Squad/website-backend/pull/735/files#diff-706d009a498138cfe882b0ee3428b8357316e116c8f4866382eb4ea1f328d6da) does this suffice?

it would be great if we could add in integration test. This expansion in coverage will enhance our confidence in the system. 😊